earth-genome/foodtwin-global-app
# Add area indicators and country code

### Contributes to: #65, #43, #20

### Changes:

- Reverted the upgrade of `npm-run-path` from v5.3.0 to v6.0.0 introduced in #62, as it was breaking the ingest script
- Pinned Prisma to v5.20.0.
- Added a new `meta` JSON property to the `Area` model to store indicators, country codes, and other metadata
- Updated the seed script to populate the `meta` property
- Connected the indicator data to the area panel
- Added country codes to the area label on the area panel

### How to Test:

- Set up the local server to use the database `global-food-twin-dev-db`
- Navigate to an area page; indicators should be displayed correctly in the side panel

### Known Issues:

- I performed a new ingestion with a reduced number of flows to speed up the process, but unfortunately, most areas don’t have flow data. Here are some example areas that do:

  - http://localhost:3000/area/DEU.10_1
  - http://localhost:3000/area/CRI.3_1
  - http://localhost:3000/area/CHN.19_1

- The country codes use 3-letter abbreviations because this is the only format available in the provided data

### Screen capture

<img width="1399" alt="Screenshot 2024-10-02 at 21 40 26" src="https://github.com/user-attachments/assets/022eaacf-e5d7-4c13-9bbf-e249fef4ce93">


@oliverroick this is ready for review.

cc @wrynearson @faustoperez
